Content/NPCs/Town/BlacksmithNPC.cs

```diff
@@ -10,12 +10,15 @@
 using PathOfTerraria.Content.Items.Gear.Weapons.Sword;
 using PathOfTerraria.Common.Utilities.Extensions;
 using PathOfTerraria.Common.NPCs;
+using Terraria.DataStructures;
 
 namespace PathOfTerraria.Content.NPCs.Town;
 
 [AutoloadHead]
-public class BlacksmithNPC : ModNPC, IQuestMarkerNPC
+public class BlacksmithNPC : ModNPC, IQuestMarkerNPC, ISpawnInRavencrestNPC
```

Comment:
Do you mind explaining what this does? Does making it loadable or something do it automatically?

Reply:
This automatically spawns it in Ravencrest (and defines where). That way I can adjust and make all of them spawn in Ravencrest without a block of NewNPC in Ravencrest gen.